spec/models/user_spec.rb

```diff
@@ -0,0 +1,33 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  subject { create :user }
+  
+  it { should validate_presence_of :username }
+  it { should validate_presence_of :email }
+  it { should validate_uniqueness_of :username }
+  it { should validate_uniqueness_of :email }
+  
+  describe "to_s" do
+    it "returns username" do
+      subject.username = "Test name"
+      expect(subject.to_s).to eq("Test name")
+    end
+  end
+
+  describe "authenticate" do
```

Comment:
It's conventional to add a `.` before class methods, and a `#` before instance methods in specs. Gives the reader of the specs more instant context into what's being tested.